cmake/CMakeLists.txt

```diff
@@ -849,8 +851,11 @@ if(BUILD_SHARED_LIBS OR PKG_PYTHON)
     find_package(Python COMPONENTS Interpreter)
   endif()
   if(Python_EXECUTABLE)
-    file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/python)
-    install(CODE "execute_process(COMMAND ${Python_EXECUTABLE} setup.py build -b ${CMAKE_BINARY_DIR}/python install --prefix=${CMAKE_INSTALL_PREFIX} --root=\$ENV{DESTDIR}/ WORKING_DIRECTORY ${LAMMPS_PYTHON_DIR})")
+    file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/python/lib)
+    file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/python/src)
+    file(COPY ${LAMMPS_SOURCE_DIR}/version.h  DESTINATION ${CMAKE_BINARY_DIR}/python/src)
+    file(COPY ${LAMMPS_PYTHON_DIR}/README ${LAMMPS_PYTHON_DIR}/pyproject.toml ${LAMMPS_PYTHON_DIR}/setup.py ${LAMMPS_PYTHON_DIR}/lammps  DESTINATION ${CMAKE_BINARY_DIR}/python/lib)
+    install(CODE "execute_process(COMMAND ${Python_EXECUTABLE} -m pip install ${CMAKE_BINARY_DIR}/python/lib --target=${CMAKE_INSTALL_PREFIX}/lib/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages)")
```

Comment:
It would be better to not hard coded the target path as some distros install mpi-enabled python modules into specific locations.

Reply:
Feel free to propose/implement some alternate way to set the destination and that also addresses the problems with `--prefix`. ...or go back to `--prefix` and forget about #3429. This is why it is flagged as a draft. My main intent was to try transfer the method of building/installing the LAMMPS python module from `make install-python` to `make install`. This is simpler, since none of the trickery to locate, copy, and bundle the pre-compiled shared library with the wheel package is required. I thought this could help to move this forward before executing `setup.py` will fail.

Reply:
Probably the only "clean" and general way of solving this is either using `Python_SITELIB` and stripping the common prefix or writing your own `execute_process` that runs a short Python snippet to compute the relative path for you.
Alternatively use what Axel is suggesting, but make the location of `--target` an advanced CMake cache variable. Something like `CMAKE_INSTALL_PYTHON_TARGET`.

Reply:
BTW: I think at this point we should perhaps stop worrying about compatibility with older CMake versions for certain targets and features and thus avoid having to program all kinds of backward compatibility hacks and instead just have gracefully degraded performance. For example in this case, there could be simply a warning message that to have the python module installed, CMake 3.12 or newer is required (which has the better way to query Python support).

Reply:
`LAMMPS_PIP_INSTALL_TARGET`, but yeah that works.

And pip has some `DESTDIR` support these days through `--root $DESTDIR --installed-root /`, see https://github.com/pypa/pip/issues/1351, but to not make our logic in CMAke overly complicated, we could just add something like `LAMMPS_PIP_EXTRA_INSTALL_OPTIONS` at the end of the command and then an expert user can inject whatever extra option incl. an overwrite for the default `--target`.

Reply:
And to Axel's other point, I am all for dropping the support for CMake<3.12 for some of these more expert feature.

Reply:
After some experiments, it appears the unconditional use of the --root flag seems to trigger the unexpected path for installing the Python module. So the hard-coded path can go away and with the latest modification it seems to be installing the LAMMPS Python module as expected. If one wants to install the python module for local use only, and not all/most of LAMMPS, people need to use "make install-python" as that also includes the shared library with the wheel package, while 'make install' depends on the installed shared library and, if needed, setting LD_LIBRARY_PATH accordingly.